torch/csrc/jit/codegen/cuda/lower_scalar_hoist.cpp

```diff
@@ -245,6 +245,27 @@ std::vector<Val*> CommonScalarMap::getHoistedScalars(kir::ForLoop* loop) const {
   return result;
 }
 
+void CommonScalarMap::initialize(const std::vector<Expr*> exprs) {
+  // We only hoist scalars not depending on tensors. In lowered expressions, all
+  // these scalars are computed in top level scope.
+  TORCH_INTERNAL_ASSERT(
+      common_scalar_map_.empty(),
+      "CommonScalarMap used before initialization.");
+  for (auto expr : exprs) {
+    if (lower_utils::isScalarExpr(expr) && expr->outputs().size() == 1) {
+      common_scalar_map_[nullptr].emplace_back(expr->output(0));
+    } else if (ir_utils::isTvOp(expr)) {
+      // We only try to reuse scalar expressions placed on the top. For example,
+      // if I have
+      // i1 = address(T0)
+      // i2 = i1 + 1
+      // Then we will not consider `i2 = i1 + 1` as a reuse-opportunity. Having
+      // this simplification helps us to keep the analysis simple.
+      break;
```

Comment:
This `break` here is because there must be no more scalar expression once we see a TV expression in the sorted expression list, right? May be a good idea to mention it as a comment.

Reply:
> there must be no more scalar expression once we see a TV expression in the sorted expression list
There might still be scalar expressions once we see a TV expression, but these scalar expressions must have a dependency on some tensor, so we give up reusing to keep our analysis simple. I just rewritten this comment: See https://github.com/csarofeen/pytorch/pull/2252#discussion_r1046601598